src/Documents.ts

```diff
@@ -829,6 +746,128 @@ class Documents implements Pointer.Library, Pointer.Validator, ObjectSchema.Reso
+		let originalFragments:PersistedFragment.Class[] = persistedDocument.getFragments();
+		let setFragments:Set<string> = new Set( originalFragments.map( fragment => fragment.id ) );
+
+		let updatedData:Pointer.Class;
+		for( let fragmentResource of fragmentResources ) {
+			updatedData = <Pointer.Class> this.compact( fragmentResource, {}, persistedDocument );
+
+			let fragment:PersistedFragment.Class = this.getAssociatedFragment( persistedDocument, updatedData );
+			if ( fragment ) {
+				fragment = this.updateObject( fragment, updatedData );
+				if ( ! persistedDocument.hasFragment( fragment.id ) ) {
+					persistedDocument.createFragment( fragment.id, fragment );
+				}
+			} else {
+				fragment = persistedDocument.createFragment( updatedData.id, updatedData );
+			}
+			setFragments.delete( fragment.id );
+
+			fragment._syncSnapshot();
+		}
+		Array.from( setFragments ).forEach( id => persistedDocument.removeFragment( id ) );
+		persistedDocument._syncSavedFragments();
+
+		updatedData = <Pointer.Class> this.compact( documentResource, {}, persistedDocument );
+		this.updateObject( persistedDocument, updatedData );
+
+		persistedDocument._syncSnapshot();
+
+		return persistedDocument;
+	}
+
+	private sendRequestForResponseDescription( uri:string, requestOptions:HTTP.Request.Options ):Promise<[ PersistedDocument.Class[], HTTP.Response.Class ]> {
+		return HTTP.Request.Service.get( uri, requestOptions, new HTTP.JSONLDParser.Class() ).then( ( [ expandedResult, response ]:[ any, HTTP.Response.Class ] ) => {
+			let freeNodes:RDF.Node.Class[] = RDF.Node.Util.getFreeNodes( expandedResult );
+			let rdfDocuments:RDF.Document.Class[] = RDF.Document.Util.getDocuments( expandedResult );
+
+			rdfDocuments.forEach( rdfDocument => this.getPersistedDocument( rdfDocument, response ) );
+			let freeResources:FreeResources.Class = this.getFreeResourcesDocument( freeNodes );
+
+			let descriptionResources:ResponseDescription.Class[] = <any> freeResources.getResources().filter( resource => ResponseDescription.Factory.hasRDFClass( resource ) );
+			if( descriptionResources.length === 0 ) return [ [], response ];
+			if( descriptionResources.length > 1 ) throw new HTTP.Errors.BadResponseError( "The response contained multiple c:ResponseDescription objects", response );
+
+			let responseDescription:ResponseDescription.Class = descriptionResources[ 0 ];
+			for ( let responseMetaData of responseDescription.responseProperties ) {
+				let document:PersistedDocument.Class = <any> responseMetaData.responsePropertyResource;
+				document._etag = responseMetaData.eTag;
+			}
+
+			let persistedDocuments:PersistedDocument.Class[] = responseDescription.responseProperties.map( ( responseMetaData:ResponseMetaData.Class ) => <any> responseMetaData.responsePropertyResource );
+			return [ persistedDocuments, response ];
+		} );
+	}
+
+	private getFreeResourcesDocument( nodes:RDF.Node.Class[] ):FreeResources.Class {
```

Comment:
Please rename it to `getFreeResources`. When we decide to rename the model to FreeResourcesDocument or something like that, we can rename everything, until then let's be consistent.